netaddr.go

```diff
@@ -423,10 +397,10 @@ func (ip IP) IsLoopback() bool {
 // value, it will return false.
 func (ip IP) IsMulticast() bool {
 	if ip.Is4() {
-		return ip.a[o4+0]&0xf0 == 0xe0
+		return uint8(ip.lo>>24)&0xf0 == 0xe0
 	}
 	if ip.Is6() {
-		return ip.a[0] == 0xff
+		return uint8(ip.hi>>56) == 0xff
```

Comment:
Ideally `return ip.v6(0) == 0xff` if inliner cooperates.

Reply:
Otherwise I'd also be fine with a method per byte we care about, just so it reads nicer at the call site here. e.g. `return ip.v6_0() == 0xff`